### PR TITLE
Nick Edwin

### PR DESCRIPTION
**Pull request by NickEdwin**

### CODE REVIEW

File: lib/school.rb [hunk 1]
```diff
@@ -1 +1,25 @@
+class School
+  attr_reader :start_time,
```
Comment (timomitchel) [inline lib/school.rb:2]: Solid use of `attr_reader`

File: lib/school.rb [hunk 2]
```diff
@@ -1 +1,25 @@
...
+              :hours_in_school_day,
+              :student_names
 
+  def initialize(start_time, hours_in_school_day)
```
Comment (timomitchel) [inline lib/school.rb:6]: Exactly what we were hoping for in `initialize`

File: lib/school.rb [hunk 3]
```diff
@@ -1 +1,25 @@
...
+    @start_time = start_time
+    @hours_in_school_day = hours_in_school_day
+    @student_names = []
+  end
+
+  def add_student_name(student)
```
Comment (timomitchel) [inline lib/school.rb:12]: Nice!

File: lib/school.rb [hunk 4]
```diff
@@ -1 +1,25 @@
...
+    @student_names << student
+  end
+
+  def end_time
+    (start_time.to_i + hours_in_school_day).to_s + ":00"
```
Comment (timomitchel) [inline lib/school.rb:17]: Really nice job with this method!

File: lib/school.rb [hunk 5]
```diff
@@ -1 +1,25 @@
...
+  end
+
+  def is_full_time?
+    if @hours_in_school_day > 4 == true
```
Comment (timomitchel) [inline lib/school.rb:21]: This is close, but we don't need the `if` statement and we don't need to compare a boolean condition to a `true` or `false` boolean literal either. This method should read:
```
def is_full_time?
    @hours_in_school_day > 4
end
```

File: test/school_test.rb [hunk 6]
```diff
@@ -45,11 +41,17 @@ def test_it_can_add_student_names
...
   def test_it_can_calculate_end_time
-    skip
     school1 = School.new('9:00', 7)
     school2 = School.new('9:00', 3)
 
     assert_equal '16:00', school1.end_time
     assert_equal '12:00', school2.end_time
   end
+
+  def test_is_full_time?
```
Comment (timomitchel) [inline test/school_test.rb:51]: TDD is not a natural process at first, but the more you practice the more it feels right. And, mod 2 is nearly impossible without TDD, so definitely keep practicing and let me know if you have specific questions. This test should be set up similarly to the previous tests:
```
# We define our test method
def test_is_full_time?
#We instantiate an object to test against
 school = School.new('9:00', 7)
#We setup our assert with the value that we expect our School instance method #is_full_time? to return. 
assert_equal true, school.is_full_time?
end
```

**Comment by NickEdwin** [comment]
Thank you for the feedback! I had really gotten stuck with the time and my
searches were leading me to a lot of gems and not stuff I could use in this
instance. I ended up with + ":00" method and questioned it but moved on,
after that the clock got the best of me! :( I have a meeting with Graham
tonight at 5, hoping it helps.

**Pull request closed without merging**